File: DAM1/T7/Trabajadores/src/model/Trabajador.java
package model;

abstract public class Trabajador {
    private String nombre, apellido;
    private int numeroSSocial, salario;

    public Trabajador() {
    }

    public Trabajador(String nombre, String apellido, int numeroSSocial, int salario) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.numeroSSocial = numeroSSocial;
        this.salario = salario;
    }

    public void mostrarDatos(){
        System.out.println("nombre = " + nombre);
        System.out.println("apellido = " + apellido);
        System.out.println("numeroSSocial = " + numeroSSocial);
        System.out.println("salario = " + salario);
    }

    public abstract void calcularSalarioMes();

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public int getNumeroSSocial() {
        return numeroSSocial;
    }

    public void setNumeroSSocial(int numeroSSocial) {
        this.numeroSSocial = numeroSSocial;
    }

    public int getSalario() {
        return salario;
    }

    public void setSalario(int salario) {
        this.salario = salario;
    }
}

File: DAM2/PMDM/T1/TrabajadoresHerencia/src/model/Trabajador.kt
package model

abstract class Trabajador(nombre : String, apellido : String, dni : String, var salario : Double) : Persona(nombre, apellido, dni){

    //nombre, apellido, dni, Salario
    //op : telefono, mail, seguro
    var seguro : Boolean = false

    constructor(nombre: String, apellido: String, dni: String, salario: Double, telefono : Int, correo : String, seguro : Boolean) : this(nombre, apellido, dni, salario) {
        this.seguro = seguro
        this.telefono = telefono
        this.correo = correo
    }

    override fun mostrarDatos() {
        super.mostrarDatos()
        println("salario = ${salario}")
        println("seguro = ${seguro}")
    }

    abstract fun calcularSalarioNeto() : Double
}